### PR TITLE
PWN-4383 - restore screen added

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4383

## Description of Work

Restore screen added

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        ::handleSignResult
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            toolbar.setNavigationOnClickListener {
+                popBackStack()
+            }
+            buttonGoogle.setOnClickListener {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt:49]: i would name button like `buttonUseGoogle` or `buttonRestoreByGoogle`
whatcha think?

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    override fun setAlternativeIdToken(userId: String, idToken: String) {
+        launch {
+            // TODO modify logic with userSignInInteractor
+            when (val result = userSignUpInteractor.trySignUpNewUser(idToken, userId)) {
+                is UserSignUpInteractor.SignUpResult.SignUpSuccessful -> {
+                    view?.onSuccessfulSignUp()
+                }
+                is UserSignUpInteractor.SignUpResult.SignUpFailed -> {
+                    Timber.e(result.cause)
+                    result.message?.let { view?.showError(it) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt:25]: showErrorSnackbar

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    override fun setAlternativeIdToken(userId: String, idToken: String) {
+        launch {
+            // TODO modify logic with userSignInInteractor
+            when (val result = userSignUpInteractor.trySignUpNewUser(idToken, userId)) {
+                is UserSignUpInteractor.SignUpResult.SignUpSuccessful -> {
+                    view?.onSuccessfulSignUp()
+                }
+                is UserSignUpInteractor.SignUpResult.SignUpFailed -> {
+                    Timber.e(result.cause)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt:24]:
```suggestion
                    Timber.e(result.cause, result.message)
```

File: app/src/main/res/layout/fragment_common_restore.xml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        app:titleTextColor="@color/night" />
+
+    <ImageView
+        android:id="@+id/imageView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="20dp"
+        android:adjustViewBounds="true"
+        android:importantForAccessibility="no"
+        android:src="@drawable/lock_image"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_common_restore.xml:30]: lock_image -> image_lock? type is always the first, isn't it?

File: app/src/main/res/layout/fragment_common_restore.xml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    <TextView
+        android:id="@+id/textViewTitle"
+        style="@style/UiKit.TextAppearance.SemiBold.LargeTitle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginHorizontal="20dp"
+        android:layout_marginTop="26dp"
+        android:gravity="center"
+        android:text="@string/restore_choose_method"
+        android:textColor="@color/night"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_common_restore.xml:46]: text_night

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner) {
+            popBackStack()
+        }
+    }
+
+    override fun startGoogleFlow() {
+        signInHelper.showSignInDialog(requireContext(), googleSignInLauncher)
+    }
+
+    override fun showError(error: String) {
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt:72]: you show have similar method in BaseMvpView

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    private val userSignUpInteractor: UserSignUpInteractor,
+) : BasePresenter<CommonRestoreContract.View>(), CommonRestoreContract.Presenter {
+
+    override fun useGoogleAccount() {
+        view?.startGoogleFlow()
+    }
+
+    override fun setAlternativeIdToken(userId: String, idToken: String) {
+        launch {
+            // TODO modify logic with userSignInInteractor
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt:18]: cool

File: app/src/main/res/layout/fragment_common_restore.xml [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@drawable/bg_night_top_rounded"
+        android:orientation="vertical"
+        android:paddingHorizontal="20dp"
+        android:paddingVertical="20dp"
+        app:layout_constraintBottom_toBottomOf="parent">
+
+        <org.p2p.uikit.components.UiKitButton
+            android:id="@+id/button_restore_by_google"
```
Comment (1sakov) [inline app/src/main/res/layout/fragment_common_restore.xml:79]: Let's use camelCase guys? 
`buttonRestoreByGoogle`? 
I can't see a real benefit from snake case here.

@elliotgrin @gslevinkov

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt (hunk 6 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestoreFragment.kt:72]: yes you are right but we need logic 
setLoadingState(isScreenLoading = false) to set in a view on error

**Pull request merged**